contracts/contracts/gatekeepers/hatsGatekeepers/HatsGatekeeperBase.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { SignUpGatekeeper } from "../SignUpGatekeeper.sol";
+import { IHats } from "../../interfaces/IHats.sol";
+
+/// @title HatsGatekeeperBase
+/// @notice Abastract contract containing the base elements of a Hats Gatekeeper contract
+abstract contract HatsGatekeeperBase is SignUpGatekeeper, Ownable {
```

Comment:
@ctrlc03 does it make sense to use interface instead of abstract contract and inherit it from SignUpGatekeeper? In terms of gas it will save more.

Reply:
@spengrah what you think here? you write much more solidity than me so probably you know the best way to go about this?

Reply:
My rationale for using an abstract contract here is that there is common logic between *Single and *Multiple, not just a common interface

Reply:
But it not *too* much common logic, so we could ~inline it into each version. 

@ctrlc03 where do the gas savings come from in your estimation?

Reply:
@spengrah before we had abstract contracts for most of the contracts, but then we decide to optimize it and migrated to interfaces due to gas savings. My idea here is to use interface, add custom errors there instead of abstract contract, but I think I'm good with your point about the common logic.

Reply:
I'd suggest we keep it as abstract, since in if we switched to interface we'd have to repeat the state vars and constants as well as the `setMaciInstance()` implementation in each of the contracts. Also, custom errors don't play very nicely with interfaces, so we'd likely need to repeat those in each contract as well.